mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivateContents.java

```diff
@@ -375,6 +375,11 @@ public Response createFileOrDirectory(InputStream contents) {
         if (mcrPath.getNameCount() > 1) {
             MCRPath parentDirectory = mcrPath.getParent();
             try {
+                if (!Files.exists(parentDirectory)
+                    && parentDirectory.toAbsolutePath().startsWith(
+                        MCRPath.getPath(derid.toString(), "/").toAbsolutePath())) {
+                    Files.createDirectories(parentDirectory);
+                }
```

Comment:
Should be:
```java
Files.createDirectories(parentDirectory);
```
It does the check for existence already and the other checks have no reason I think. Check the call to createDirectories() for a FileAlreadyExistsException if the parent directory is a file to return the right error message later in the code